core/src/apps/bitcoin/sign_tx/progress.py

```diff
@@ -1,31 +1,129 @@
-from trezor import ui, utils, workflow
+from typing import TYPE_CHECKING
```

Comment:
As the size of this file increased a lot (from 30 to 130 lines), it might start making sense to create a `Progress` class instead of this collection of functions, that need to use the `global` keyword quite a lot.

(The `ProgressObj` would need to be created here, so it can be shared for all the places that call it.)

From size-perspective creating a class could be more costly, but the readability should be better. (And potentially modifying the instance variables could be less expensive than modifying global variables, which is happening quite often here.)

Reply:
https://github.com/trezor/trezor-firmware/pull/2507/commits/236640251228724ca3db76c8f3293fd33477d33c